src/btree/bt_read.c

```diff
@@ -556,12 +553,14 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
 	 *
 	 * Don't free WT_REF.page_las, there may be concurrent readers.
 	 */
-	if (final_state == WT_REF_MEM &&
-	    ref->page_las != NULL && (prepared || !ref->page_las->skew_newest))
-		WT_ERR(__wt_las_remove_block(
-		    session, ref->page_las->las_pageid));
+	if (final_state == WT_REF_MEM && ref->page_las != NULL &&
+	    (!ref->page_las->skew_newest || ref->page_las->has_prepares)) {
+		ret = __wt_las_remove_block(
+		    session, ref->page_las->las_pageid);
+		WT_ERR(ret);
```

Comment:
Was there other code you temporarily had here that the call was split into `ret =` and `WT_ERR(ret)`? It can be put back together.

Reply:
Yes, I was asserting that this call didn't fail (and didn't see it fail with the test cases I tried).  I agree this part should be reverted.